### PR TITLE
[Platformer] Exclude "potential colliding" platforms that are not actually in the searched area.

### PR DESCRIPTION
**Pull request by D8H**
This filters platforms that are not in the searched area anymore. It can happen because platforms are not updated in the RBush before that characters movement are being processed.

The tests pass when merge with:
* https://github.com/4ian/GDevelop/pull/3376

It won't make it up for platforms that should have been found by RBush. So, this PR still stand:
* https://github.com/4ian/GDevelop/pull/2602

### CODE REVIEW

File: Extensions/PlatformBehavior/platformruntimebehavior.ts [hunk 1]
```diff
@@ -88,6 +88,26 @@ namespace gdjs {
       result.length = 0;
       result.push.apply(result, nearbyPlatforms);
     }
+
+    /**
+     * Check if the platforms is around the specified object.
+     * @param maxMovementLength The maximum distance, in pixels, the object is going to do.
+     * @return true if the platforms is around the specified object.
+     */
+    isPlatformsAround(
```
Comment (4ian) [inline Extensions/PlatformBehavior/platformruntimebehavior.ts:97]:
```suggestion
    isPlatformAround(
```

Can this be static?

File: Extensions/PlatformBehavior/platformruntimebehavior.ts [hunk 2]
```diff
@@ -88,6 +88,26 @@ namespace gdjs {
       result.length = 0;
       result.push.apply(result, nearbyPlatforms);
     }
+
+    /**
+     * Check if the platforms is around the specified object.
```
Comment (4ian) [inline Extensions/PlatformBehavior/platformruntimebehavior.ts:93]:
```suggestion
     * Check if the platform is around the specified object.
```

File: Extensions/PlatformBehavior/platformruntimebehavior.ts [hunk 3]
```diff
@@ -88,6 +88,26 @@ namespace gdjs {
...
+     * @param maxMovementLength The maximum distance, in pixels, the object is going to do.
+     * @return true if the platforms is around the specified object.
```
Comment (4ian) [inline Extensions/PlatformBehavior/platformruntimebehavior.ts:95]:
```suggestion
     * @return true if the platform is around the specified object.
```

File: Extensions/PlatformBehavior/platformerobjectruntimebehavior.ts [hunk 4]
```diff
@@ -965,17 +965,26 @@ namespace gdjs {
      * Update _potentialCollidingObjects member with platforms near the object.
      */
     private _updatePotentialCollidingObjects(maxMovementLength: float) {
+      const object = this.owner;
+
       this._manager.getAllPlatformsAround(
```
Comment (4ian) [inline Extensions/PlatformBehavior/platformerobjectruntimebehavior.ts:970]: Question: should the filtering be done by `getAllPlatformsAround`? Or by the RBush somehow?

**Review by 4ian — APPROVED** [review]

File: Extensions/PlatformBehavior/platformerobjectruntimebehavior.ts (hunk 4 above)
Comment (D8H) [reply Extensions/PlatformBehavior/platformerobjectruntimebehavior.ts:970]: Yes, I moved it in `getAllPlatformsAround`.
RBush do what it is asked. It's not its fault it has outdated AABB.

Comment (4ian) [reply Extensions/PlatformBehavior/platformerobjectruntimebehavior.ts:970]: Perfect, thanks!

File: Extensions/PlatformBehavior/platformruntimebehavior.ts [hunk 5]
```diff
@@ -73,20 +74,31 @@ namespace gdjs {
...
-      // @ts-ignore
       searchArea.maxY = y + oh / 2 + maxMovementLength;
       const nearbyPlatforms = this._platformRBush.search(searchArea);
+
       result.length = 0;
-      result.push.apply(result, nearbyPlatforms);
+      for (let i = 0; i < nearbyPlatforms.length; i++) {
+        const platform = nearbyPlatforms[i];
+        const platformAABB = platform.owner.getAABB();
+        const platformIsStillAround =
```
Comment (4ian) [inline Extensions/PlatformBehavior/platformruntimebehavior.ts:90]: Very clear with this name 👍

**Review by 4ian — APPROVED** [review]

**Comment by AlexandreSi** [comment]
Merging your PR in order to merge #3376 ! Thank you again for your help!

**Pull request merged**